samples/snippets/src/test/java/pubsublite/beam/BeamReadWriteIT.java

```diff
@@ -0,0 +1,189 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pubsublite.beam;
+
+import static com.google.common.truth.Truth.assertThat;
+import static junit.framework.Assert.fail;
+import static junit.framework.TestCase.assertNotNull;
+
+import com.google.cloud.pubsublite.proto.SequencedMessage;
+import com.google.common.collect.ImmutableList;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.UUID;
+import javax.annotation.concurrent.GuardedBy;
+import org.apache.beam.runners.direct.DirectOptions;
+import org.apache.beam.runners.direct.DirectRunner;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.options.StreamingOptions;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.SimpleFunction;
+import org.apache.beam.sdk.values.PCollection;
+import org.joda.time.Duration;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import pubsublite.CreateSubscriptionExample;
+import pubsublite.CreateTopicExample;
+import pubsublite.DeleteSubscriptionExample;
+import pubsublite.DeleteTopicExample;
+
+public class BeamReadWriteIT {
+  private ByteArrayOutputStream bout;
+  private PrintStream out;
+  Random rand = new Random();
+
+  private static final Long projectNumber =
+      Long.parseLong(System.getenv("GOOGLE_CLOUD_PROJECT_NUMBER"));
+  private String cloudRegion = "us-central1";
+  private final char zoneId = (char) (rand.nextInt(3) + 'a');
+  private static final String suffix = UUID.randomUUID().toString();
+  private static final String topicId = "lite-topic-" + suffix;
+  private static final String subscriptionId = "lite-subscription-" + suffix;
+  private static final int partitions = 2;
+  private static final int messageCount = 50;
+
+  private static void requireEnvVar(String varName) {
+    assertNotNull(
+        "Environment variable " + varName + " is required to perform these tests.",
+        System.getenv(varName));
+  }
+
+  @Rule public Timeout globalTimeout = Timeout.seconds(300); // 5 minute timeout
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_CLOUD_PROJECT_NUMBER");
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    bout = new ByteArrayOutputStream();
```

Comment:
Can you add a comment explaining why this is necessary? Why are you printing that the default PrintStream cannot handle?

Reply:
Nack. This is the same pattern followed by all other samples. If it needs a comment here, it would need a comment there. I have no idea why PrintStream isn't used.